base/sort.jl

```diff
@@ -603,19 +640,26 @@ function _sort!(v::AbstractVector, a::IEEEFloatOptimization, o::Ordering, kw)
         j = send_to_end!(x -> after_zero(o, x), v; lo, hi)
         scratch = _sort!(iv, a.next, Reverse, (;kw..., lo, hi=j))
         if scratch === nothing # Union split
-            _sort!(iv, a.next, Forward, (;kw..., lo=j+1, hi, scratch))
+            _sort!(iv, a.next, Forward, (;kw..., lo=j+1, hi, nothing))
         else
             _sort!(iv, a.next, Forward, (;kw..., lo=j+1, hi, scratch))
```

Comment:
@LilithHafner Btw, is `scratch` type-stable here, i.e., can compiler infer that `scratch` cannot be `Nothing`?

Reply:
I recall that removing the if statement entirely introduced dynamic dispatch. IICU, all of type inference is an implementation detail, so I'm not totally sure, but I believe that we need an if statement to force union splitting though it works well whether we use `_sort!(..., nothing)` or `_sort!(..., scratch)` because the compiler can determine the type of `scratch` at compile time..